### PR TITLE
Tnt landmines

### PR DESCRIPTION
**Pull request by kruthers**
Added TNT Landmines

### CODE REVIEW

**Comment by sparks0** [comment]
@kruthers could you descrive a list of new features or changes to mechanics in the PR description?

File: gm4_tnt_landmines/data/tnt_landmines/functions/upgrade.mcfunction [hunk 1]
```diff
@@ -0,0 +1,3 @@
+#@s=updage Item
+kill @s
+tag @e[type=armor_stand,tag=gm4_tnt_mine,distance=..0.2] add noDmg
```
Comment (sparks0) [inline gm4_tnt_landmines/data/tnt_landmines/functions/upgrade.mcfunction]: should be gm4_no_damage or gm4_no_dmg

File: gm4_tnt_landmines/data/tnt_landmines/functions/pulse_check.mcfunction [hunk 2]
```diff
@@ -0,0 +1,6 @@
+#compares assigned run tick to current tick and calls main
+execute if score gm4_currentTick gm4_clockTick = tnt_landmines gm4_clockTick run function tnt_landmines:main
```
Comment (sparks0) [inline gm4_tnt_landmines/data/tnt_landmines/functions/pulse_check.mcfunction]: gm4_currentTick is now gm4_current_tick
gm4_clockTick is now gm4_clock_tick

File: gm4_tnt_landmines/data/tnt_landmines/functions/init_check.mcfunction [hunk 3]
```diff
@@ -0,0 +1,3 @@
+#unless the module is already initialized
+execute unless score tnt_landmines gm4_modules matches 1.. run function tnt_landmines:init
+scoreboard players add installedModules gm4_upCheck 1
```
Comment (sparks0) [inline gm4_tnt_landmines/data/tnt_landmines/functions/init_check.mcfunction]: installedModules is now installed_modules
gm4_upCheck is now gm4_up_check

File: gm4_tnt_landmines/data/tnt_landmines/functions/init.mcfunction [hunk 4]
```diff
@@ -0,0 +1,18 @@
+#announce module installation
+tellraw @a[gamemode=creative] ["",{"text":"[GM4]: Installing TNT Landmines..."}]
+execute unless entity @p run say GM4: Installing TNT Landmines...
+
+#declare and initialise scoreboards and settings
+scoreboard players set updateHappened gm4_upCheck 1
```
Comment (sparks0) [inline gm4_tnt_landmines/data/tnt_landmines/functions/init.mcfunction]: updateHappened is now update_happened

File: gm4_tnt_landmines/data/gm4/functions/tick.mcfunction [hunk 5]
```diff
@@ -0,0 +1,9 @@
+#runs every tick
+scoreboard players add gm4_currentTick gm4_clockTick 1
+execute if score gm4_currentTick gm4_clockTick matches 16.. run scoreboard players set gm4_currentTick gm4_clockTick 0
+function #gm4:pulse_check
+function #gm4:repeat
+
+#prints scheduled out-of-date messages to chat
+execute if score gm4_currentTick gm4_clockTick matches 0 if entity @a[limit=1,gamemode=creative] if entity @e[limit=1,tag=message,type=area_effect_cloud] run tellraw @a ["",{"text":"The following modules are out of date: ","color":"aqua"},{"selector":"@e[type=area_effect_cloud,tag=message]","color":"gold","insertion":"/tellraw @p %s"}]
```
Comment (sparks0) [inline gm4_tnt_landmines/data/gm4/functions/tick.mcfunction]: gm4_current_tick is now current_tick

**Comment by kruthers** [comment]
@sparks0 Made the changes you requested as well as updating it to the new clock (i think, at least in my testing it worked) and the list of all the changes from the original tnt landmines module:

1) They are all crafted in custom crafters now.
- For the base landmine the recipe is: row 1 is [grass block, stone pressure plate ,grass block], row 2 is [dirt, tnt,dirt] and row 3 is [air, redstone dust, air]
- For the disguised ones it is very similar except the grass blocks will go where the dirt blocks are and where the grass blocks used to be will be where the block they are disguising as will go.
- Possible disguises: grass, dirt, stone, stone bricks, crafting table and netherrack. They is the invisible option if diamonds are used
2) they can be upgraded to make is so they summon an area effect cloud instead of blowing up the terrain. 
- This is crafted in a custom crafter with the following recipe: row 1 is [potion, stone pressure plate potion] row 2 is [obsidian, tnt, obsidian]. 
- The potions depend on whether zauber cauldrons is installed or not, if it is it requires an instant damage 4 potion, if it is not installed it requires only an instant damage 2 potion.
3) the ability to disarm them was added, this is done by dropping a set of shears on them
- if it has silk touch on the shears it will drop items when disarming them, if not it will not

I think that is all the changes, they are still denoted by walking over them or removing the block they are in as before.

**Comment by sparks0** [comment]
Any update on this?

**Comment by sparks0** [comment]
Closed as inactive. You can reopen this PR if you make changes.

**Pull request closed without merging**